eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ThrowFromClientLoggerCheck.java

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * To throw an exception,
+ * <ol>
+ *   <li>Must do it through a 'logger.logAndThrow', rather than by directly calling 'throw exception'</li>
+ *   <li>Always call return after calling 'logger.logAndThrow'.</li>
+ * </ol>
+ *
+ * Skip check if throwing exception from
+ * <ol>
+ *   <li>Static method</li>
+ *   <li>Static class</li>
+ *   <li>Constructor</li>
+ * </ol>
+ */
+public class ThrowFromClientLoggerCheck extends AbstractCheck {
+    private static final String LOGGER_LOG_AND_THROW = "logger.logAndThrow";
+
+    private Deque<Boolean> classStaticDeque = new ArrayDeque<>();
+    private Deque<Boolean> methodStaticDeque = new ArrayDeque<>();
```

Comment:
Can you provide some commentary in the code about why you need these data structures.

Reply:
will do